cluster-autoscaler/proposals/metrics.md

```diff
@@ -21,17 +21,18 @@ All the metrics are prefixed with `cluster_autoscaler_`.
-| nodes_count | Gauge | `state`=&lt;node-state&gt; | Number of nodes in cluster. |
-| unschedulable_pods_count | Gauge | | Number of unschedulable ("Pending") pods in the cluster. |
-| node_groups_count | Gauge | `node_group_type`=&lt;node-group-type&gt; | Number of node groups managed by CA. |
-| max_nodes_count | Gauge | | Maximum number of nodes in all node groups. |
-| cluster_cpu_current_cores | Gauge | | | Current number of cores in the cluster, minus deleting nodes. |
-| cpu_limits_cores | Gauge | `direction`=&lt;`minimum` or `maximum`&gt; | Minimum and maximum number of cores in the cluster. |
-| cluster_memory_current_bytes | Gauge | | Current number of bytes of memory in the cluster, minus deleting nodes. |
-| memory_limits_bytes | Gauge | `direction`=&lt;`minimum` or `maximum`&gt; | Minimum and maximum number of bytes of memory in cluster. |
+| Metric name | Metric type | Labels                                     | Description                                                                        |
+| ----------- | ----------- |--------------------------------------------|------------------------------------------------------------------------------------|
+| cluster_safe_to_autoscale | Gauge |                                            | Whether or not cluster is healthy enough for autoscaling. 1 if it is, 0 otherwise. |
+| nodes_count | Gauge | `state`=&lt;node-state&gt;                 | Number of nodes in cluster.                                                        |
+| unschedulable_pods_count | Gauge |                                            | Number of unschedulable ("Pending") pods in the cluster.                           |
+| node_groups_count | Gauge | `node_group_type`=&lt;node-group-type&gt;  | Number of node groups managed by CA.                                               |
+| max_nodes_count | Gauge |                                            | Maximum number of nodes in all node groups.                                        |
+| cluster_cpu_current_cores | Gauge |                                            |                                                                                    | Current number of cores in the cluster, minus deleting nodes. |
+| cpu_limits_cores | Gauge | `direction`=&lt;`minimum` or `maximum`&gt; | Minimum and maximum number of cores in the cluster.                                |
+| cluster_memory_current_bytes | Gauge |                                            | Current number of bytes of memory in the cluster, minus deleting nodes.            |
+| memory_limits_bytes | Gauge | `direction`=&lt;`minimum` or `maximum`&gt; | Minimum and maximum number of bytes of memory in cluster.                          |
+ | node_taints_count | Gauge | `type`=&lt;taint-type&gt;                  | Number of taints per type in cluster.                                              |
```

Comment:
This is a pretty old proposal, not sure if there's much value in adding the metric here. There's already a lot of metrics that were added in the code but not to this doc.

Reply:
Removed